src/Compilers/VisualBasic/Portable/Lowering/AsyncRewriter/AsyncRewriter.AsyncMethodToClassRewriter.Await.vb

```diff
@@ -28,7 +28,7 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
                 ' The awaiter temp facilitates EnC method remapping and thus have to be long-lived.
                 ' It transfers the awaiter objects from the old version of the MoveNext method to the New one.
                 Debug.Assert(node.Syntax.IsKind(SyntaxKind.AwaitExpression))
-                Dim awaiterType As TypeSymbol = node.GetAwaiter.Type.InternalSubstituteTypeParameters(Me.TypeMap)
+                Dim awaiterType As TypeSymbol = node.GetAwaiter.Type.InternalSubstituteTypeParameters(Me.TypeMap).AsTypeSymbol()
```

Comment:
The assertion in AsTypeSymbol could fail if the library's GetAwaiter() method has a custom modifier. Is an assertion failure the behavior we want?